File: x402-demo-server/scratch/test_groq.ts
import Groq from "groq-sdk";
import dotenv from "dotenv";
import path from "path";

dotenv.config({ path: path.join(__dirname, "../.env") });

async function testGroq() {
  const keys = [
    process.env.GROQ_API_KEY_1,
    process.env.GROQ_API_KEY_2,
    process.env.GROQ_API_KEY_3,
  ].filter(Boolean) as string[];

  if (!keys.length) {
    console.error("No API keys found");
    return;
  }

  const client = new Groq({ apiKey: keys[0] });
  try {
    const completion = await client.chat.completions.create({
      messages: [{ role: "user", content: "Hello, reply with only one word 'success'." }],
      model: "openai/gpt-oss-120b",
      max_tokens: 10,
    });
    console.log("Full response:", JSON.stringify(completion, null, 2));
  } catch (err: any) {
    console.error("Error:", err);
  }
}

testGroq();

File: interview_pro/backend/test_groq.py
import os
import json
from dotenv import load_dotenv
from groq import Groq

load_dotenv()
keys = [os.getenv(f"GROQ_API_KEY_{i}") for i in range(20, 28) if os.getenv(f"GROQ_API_KEY_{i}")]
print(f"Loaded {len(keys)} keys from GROQ_API_KEY_20 to 27")
if not keys:
    keys = [os.getenv("GROQ_API_KEY")]

client = Groq(api_key=keys[0])
res = client.chat.completions.create(
    messages=[
        {"role": "system", "content": "You are a career coach. Output valid JSON only."},
        {"role": "user", "content": "Analyze candidate with Python, React experience for Fullstack Developer role. Return JSON with resumeMatchScore (int), skillGaps (list of str), existingSkills (list of str)."}
    ],
    model="openai/gpt-oss-120b",
    response_format={"type": "json_object"}
)
print("Response preview:")
print(res.choices[0].message.content[:300])
